# HG changeset patch
# User Yuya Nishihara <yuya@tcha.org>
# Date 1362382324 -32400
# Branch stable
# Node ID a90d30ce513a932b03c46ae063839db1e37b7ea7
# Parent  3762eb25f7f2a1ce73ba98ab00c28967fb11f0b3
reporegistry, repoview: use lightweight comparison for PYQT_VERSION

The doc says PYQT_VERSION and QT_VERSION are hexadecimal integer in
0xMMNNPP (M: major, N: minor, P: patch) format.

diff -r 3762eb25f7f2a1ce73ba98ab00c28967fb11f0b3 -r a90d30ce513a932b03c46ae063839db1e37b7ea7 tortoisehg/hgqt/reporegistry.py
--- a/tortoisehg/hgqt/reporegistry.py	Sat Mar 02 16:26:27 2013 -0600
+++ b/tortoisehg/hgqt/reporegistry.py	Mon Mar 04 16:32:04 2013 +0900
@@ -45,7 +45,7 @@
         self.setAcceptDrops(True)
         self.setAutoScroll(True)
         self.setDragDropMode(QAbstractItemView.DragDrop)
-        if PYQT_VERSION_STR.split('.') >= ['4', '7']:
+        if PYQT_VERSION >= 0x40700:
             self.setDefaultDropAction(Qt.MoveAction)
         self.setDropIndicatorShown(True)
         self.setEditTriggers(QAbstractItemView.DoubleClicked
diff -r 3762eb25f7f2a1ce73ba98ab00c28967fb11f0b3 -r a90d30ce513a932b03c46ae063839db1e37b7ea7 tortoisehg/hgqt/repotreemodel.py
--- a/tortoisehg/hgqt/repotreemodel.py	Sat Mar 02 16:26:27 2013 -0600
+++ b/tortoisehg/hgqt/repotreemodel.py	Mon Mar 04 16:32:04 2013 +0900
@@ -33,7 +33,7 @@
     xw.writeEndDocument()
 
 def readXml(source, rootElementName):
-    if PYQT_VERSION_STR.split('.') < ['4', '7']:
+    if PYQT_VERSION < 0x40700:
         return
     itemread = None
     xr = QXmlStreamReader(source)
diff -r 3762eb25f7f2a1ce73ba98ab00c28967fb11f0b3 -r a90d30ce513a932b03c46ae063839db1e37b7ea7 tortoisehg/hgqt/repoview.py
--- a/tortoisehg/hgqt/repoview.py	Sat Mar 02 16:26:27 2013 -0600
+++ b/tortoisehg/hgqt/repoview.py	Mon Mar 04 16:32:04 2013 +0900
@@ -63,7 +63,7 @@
         self.htmlDelegate = htmldelegate.HTMLDelegate(self)
 
         self.setAcceptDrops(True)
-        if PYQT_VERSION_STR.split('.') >= ['4', '7']:
+        if PYQT_VERSION >= 0x40700:
             self.setDefaultDropAction(Qt.MoveAction)
         self.setDragEnabled(True)
         self.setDropIndicatorShown(True)
         
         # HG changeset patch
# User Steve Borho <steve@borho.org>
# Date 1362690570 21600
# Branch stable
# Node ID e571f9b3eadc2e2706bd350dab33d3bee5bf2315
# Parent  6fbbe3d1974c267e5a30e4d9d1efc54a24fd1f35
qt: use int comparison for Qt and PyQt version check (fixes #2467)

diff -r 6fbbe3d1974c267e5a30e4d9d1efc54a24fd1f35 -r e571f9b3eadc2e2706bd350dab33d3bee5bf2315 tortoisehg/hgqt/qtlib.py
--- a/tortoisehg/hgqt/qtlib.py	Mon Mar 04 17:00:24 2013 -0600
+++ b/tortoisehg/hgqt/qtlib.py	Thu Mar 07 15:09:30 2013 -0600
@@ -25,9 +25,8 @@
 from PyQt4.QtCore import *
 from PyQt4.QtGui import *
 
-if PYQT_VERSION_STR.split('.') < ['4', '6'] or \
-   QT_VERSION_STR.split('.') < ['4', '6']:
-    sys.stderr.write('TortoiseHg requires Qt 4.6 and PyQt 4.7\n')
+if PYQT_VERSION < 0x40600 or QT_VERSION < 0x40600:
+    sys.stderr.write('TortoiseHg requires Qt 4.6 and PyQt 4.6\n')
     sys.stderr.write('You have Qt %s and PyQt %s\n' %
                      (QT_VERSION_STR, PYQT_VERSION_STR))
     sys.exit()
